File: 78.subsets.py
class Solution:
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        ans = [[]]
        for i in nums:
            for j in range(len(ans)):
                a = ans[j][:]
                a.append(i)
                ans.append(a)
        return ans

File: 78.subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        ans.push_back({});
        for (int i = 0; i < nums.size(); i++) {
            int n = ans.size();
            for (int j = 0; j < n; j++) {
                auto a = ans[j];
                a.push_back(nums[i]);
                ans.push_back(a);
            }
        }
        return ans;
    }
};